source/solvers/flow_control.cc

```diff
@@ -111,14 +111,36 @@ FlowControl<dim>::calculate_beta(const std::pair<double, double> &flow_rate,
   flow_rate_1n = flow_rate_n;
 }
 
-
 template <int dim>
-Tensor<1, dim>
-FlowControl<dim>::get_beta()
+void
+FlowControl<dim>::save(std::string prefix)
 {
-  return beta;
+  std::string   filename = prefix + ".flowcontrol";
+  std::ofstream output(filename.c_str());
+  output << "Flow control" << std::endl;
+  output << "Previous_beta " << beta_n << std::endl;
+  output << "Previous_flow_rate " << flow_rate_1n << std::endl;
+  output << "No_force " << no_force << std::endl;
+  output << "Threshold_factor " << threshold_factor << std::endl;
 }
 
+template <int dim>
+void
+FlowControl<dim>::read(std::string prefix)
+{
+  std::string   filename = prefix + ".flowcontrol";
+  std::ifstream input(filename.c_str());
+  if (!input)
+    {
+      throw("Unable to open file");
+    }
+  std::string buffer;
+  std::getline(input, buffer);
+  input >> buffer >> beta_n;
+  input >> buffer >> flow_rate_1n;
+  input >> buffer >> no_force;
+  input >> buffer >> threshold_factor;
```

Comment:
I always forget if it is better to close the file or if the end of the scope of the function will close it automatically anyway....

Reply:
"Note that any open file is automatically closed when the ifstream object is destroyed." :D